pkg/scaling/scale_handler.go

```diff
@@ -379,17 +376,17 @@ func (h *scaleHandler) performGetScalersCache(ctx context.Context, key string, s
 	default:
 	}
 
-	// Scalers Close() could be impacted by timeouts, blocking the mutex
-	// until the timeout happens. Instead of locking the mutex, we take
-	// the old cache item and we close it in another goroutine, not locking
-	// the cache: https://github.com/kedacore/keda/issues/5083
-	if regenerateCache {
-		oldCache := h.scalerCaches[key]
+	h.scalerCachesLock.Lock()
+	defer h.scalerCachesLock.Unlock()
+
+	if oldCache, ok := h.scalerCaches[key]; ok {
```

Comment:
Cache regeneration needs to be protected by a mutex entirely. Even the lookup of the old entry to close it. This way the old entry is now closed and updated in the same mutex block so that no scaler can leak from the cache.

I think there are to way to handle mutual exclusion in this method. There could be a single mutex usage but I think there are some network calls to kube API in this method that might not be ideal.

This implementation has 2 mutex usage so it is not blocking the mutex during IO operation. However, it is still possible for multiple goroutines to enter the method and start updating the same scaler. This should be fine as all these scaler will be closed properly but might a bit wasteful if some scalers are expensive to create.